### PR TITLE
mcs: use separate participant

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Ref #5839.

### What is changed and how does it work?

Previously, we used tso participant for all micro-services which is not suitable. This PR distinguishes them by using different proto definitions.

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/member/participant.go [hunk 1]
```diff
@@ -131,27 +146,36 @@ func (m *Participant) GetLeaderID() uint64 {
...
+func (m *Participant) setLeader(member participant) {
 	m.leader.Store(member)
 	m.lastLeaderUpdatedTime.Store(time.Now())
 }
 
 // unsetLeader unsets the member's leader.
 func (m *Participant) unsetLeader() {
-	m.leader.Store(&tsopb.Participant{})
+	var leader participant
+	switch m.serviceName {
```
Comment (JmPotato) [inline pkg/member/participant.go:170]: Will generic type eliminate these switches?

**Review by JmPotato — APPROVED** [review]

File: pkg/member/participant.go (hunk 1 above)
Comment (rleungx) [reply pkg/member/participant.go:170]: Maybe rewrite it later.

**Review by JmPotato — APPROVED** [review]

**Comment by rleungx** [comment]
@lhy1024 PTAL

File: pkg/member/participant.go [hunk 2]
```diff
@@ -22,28 +22,42 @@ import (
...
 	"github.com/tikv/pd/pkg/utils/etcdutil"
 	"go.etcd.io/etcd/clientv3"
 	"go.uber.org/zap"
 )
 
 type leadershipCheckFunc func(*election.Leadership) bool
 
+type participant interface {
+	GetName() string
+	GetId() uint64
```
Comment (lhy1024) [inline pkg/member/participant.go:41]: How about GetID

**Review by lhy1024 — APPROVED** [review]

File: pkg/member/participant.go (hunk 2 above)
Comment (rleungx) [reply pkg/member/participant.go:41]: No, it's generated by protobuf.

**Pull request merged**